src/main/java/jenkins/plugins/git/GitToolChooser.java

```diff
@@ -70,13 +70,18 @@ public GitToolChooser(String remoteName, Item projectContext, String credentials
         currentNode = n;
         this.listener = listener;
         implementation = "NONE";
-        useCache = decideAndUseCache(remoteName);
 
-        if (useCache) {
-            implementation = determineSwitchOnSize(sizeOfRepo, gitExe);
-        } else {
-            decideAndUseAPI(remoteName, projectContext, credentialsId, gitExe);
+        // Skip expensive checks if there is only one available implementation to choose from
+        if (JGIT_SUPPORTED) {
+            useCache = decideAndUseCache(remoteName);
+
+            if (useCache) {
+                implementation = determineSwitchOnSize(sizeOfRepo, gitExe);
+            } else {
+                decideAndUseAPI(remoteName, projectContext, credentialsId, gitExe);
+            }
```

Comment:
I added the following logging statement so that I could confirm the file system size checks were not performed.  As far as I can tell, they are performed whether or not the user has enabled JGit.  I think we need more code changes if we want to prevent the use of JGit unless the user has enabled JGit.
```suggestion
            }
        } else {
            LOGGER.log(Level.FINE, "JGit not supported, skipped file system checks");
``````

Reply:
> I think we need more code changes if we want to prevent the use of JGit unless the user has enabled JGit.

Err... You want to say that JGit is currently used even if user didn't enable it? What does 'enabling JGit' do at all then?